### DISCUSSION TITLE
[Discussion] Issue labels

### DISCUSSION THREAD

**Discussion by jsimnz** [sourcenetwork/defradb#433] [General]
Ive started defining some issue labels to use to better organize the outstanding issues. I wanted to open up a discussion thread to get everyones input on the newly defined ones, if they make sense, need refinement, etc. As well as suggestions for other issue labels.

Current labels
=========
`area/api`
*Related to the external API component*
 
`area/auth`
*Related to the authorization and authentication of data*
 
`area/collections`
*Related to the collections system*
 
`area/crdt`
*Related to the (Merkle) CRDT system*
 
`area/datastore`
*Related to the datastore / storage engine system*
 
`area/db-system`
*Related to the core system related components of the DB*
 
`area/errors`
*Related to the internal management or design of our error handling system*
 
`area/parser`
*Related to the parser components*
 
`area/query`
*Related to the query component*
 
`area/schema`
*Related to the schema system*
 
`area/testing`
*Related to any test or testing suite*
 
`area/verification`
*Related to the data verification system*
 
`bug`
*Something isn't working*
 
`ci/build`
*This is issue is about the build or CI system, and the administration of it.*
 
`discussion`
*Discussion thread*
 
`documentation`
*Improvements or additions to documentation*
 
`duplicate`
*This issue or pull request already exists*
 
`epic`
*Includes sub-issues*
 
`feature`
*New feature or request*
 
`good first issue`
*Good for newcomers*
 
`help wanted`
*Extra attention is needed*
 
`invalid`
*This doesn't seem right*
 
`meta/dev`
*These issues related to the meta elements of the code base like dev flow, repo logistics, and PM*
 
`perf`
*Performance issue or suggestion*
 
`question`
*Further information is requested*
 
`refactor`
*This issue requires refactoring of existing codebases and components*
 
`todo 🗒️`
*Automatically generated by* `// @todo` comments in code*
 
`wontfix`
*This will not be worked on*

**Comment by jsimnz** [comment]
Added another:
`area/logging`
*Related to the logging/logger system*

**Comment by pradhanashutosh** [comment]
Issue status labels can be:

`open`

`in progress`

`under review`

`merged`

**Comment by jsimnz** [comment]
> Issue status labels can be:
> 
> `open`
> 
> `in progress`
> 
> `under review`
> 
> `merged`

The status is handled by the "Projects" board, but ill make sure those labels are automatically updated from the board to the issue

**Comment by AndrewSisley** [comment]
If it is not too late for comment...

I like the use of `merged` (is super clear vs the normal `done`), and the `area` namespacing - what do you think of extending that style to cover the others?  E.g. `type(?)/bug`, `type/perf`, `action(??)/help-wanted`, `action/wont-fix`, etc.  We wont cover every label at first and will need to add/maintain these, and some further structure could help organise them (or possibly create additional risk of arguing/nit-picking...)

Am happy either way (I don't see this as being too important to spend much time on), just throwing the thought out there :)

**Comment by jsimnz** [comment]
> If it is not too late for comment...
> 
> I like the use of `merged` (is super clear vs the normal `done`), and the `area` namespacing - what do you think of extending that style to cover the others? E.g. `type(?)/bug`, `type/perf`, `action(??)/help-wanted`, `action/wont-fix`, etc. We wont cover every label at first and will need to add/maintain these, and some further structure could help organise them (or possibly create additional risk of arguing/nit-picking...)
> 
> Am happy either way (I don't see this as being too important to spend much time on), just throwing the thought out there :)

Yupp, more just an open-ended discussion for anyone to chime in with their thoughts.

I like extending the idea of `area` to others. I think some are self describing, but others, like `action` are well suited for that.

**Comment by shahzadlone** [comment]
perhaps a `code-freeze` to indicate an issue related to our recent code freeze, incase we adopt the methodology.

**Reply by orpheuslummis** [reply]
@shahzadlone what do you think about this now?

**Reply by shahzadlone** [reply]
I don't see much value for this unless someone has a good use case?
